src/transformers/models/atlas/retriever.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
We have a 1 model 1 file policy so this should be part of the `modeling_file`.  There should also be no dependencies with other model, but rather a use of the `# Copied from` statements. See [here](https://huggingface.co/blog/transformers-design-philosophy)